src/FSharp.Core/prim-types.fs

```diff
@@ -369,6 +369,11 @@ namespace Microsoft.FSharp.Core
     type ValueAsStaticPropertyAttribute() =
         inherit Attribute()
 
+    [<AttributeUsage(AttributeTargets.Method ||| AttributeTargets.Property, AllowMultiple=false)>]
+    [<Sealed>]
+    type NoFscInliningAttribute() =
```

Comment:
Are we set on the attribute name? :)
Maybe something like `NoCompilerInlining` (though JIT is also a compiler)...

Reply:
Whatever you want. I treated it as a placeholder.

Reply:
I would vote something like "NoCompilerInlining", not sure what others think.

Reply:
I would want to be more explicit about what it really does.
The term "compiler" os overloaded.

Mazbe "NoFsharpCompilerInlining"

Reply:
Ping me with the final name and I'll rename everything. Theren's nothing else that needs doing that I can see.

Reply:
I am fine with both NoFscInliningAttribute as well as NoFsharpCompilerInlining.

@vzarytovskii  ?

Reply:
Naming is one of the hardest problems in computer science.

I don't personally like the fsharp or FSC in the name, and prefer more generalized ones.

But it's a personal preference, forks for me if you're okay with it.

Reply:
The XML doc summary should be very clear in any case.

Reply:
NoCompilerInlining is probably least wierd.

This is presumably expected to operate alongside the MethodImplAttribute.Inline field.  The developer who applies it will be expected to understand the subtlety of Jit inlining and fsc code inlining.

For sure, it's gonna be wierd to express inlining and no inlining at the same time :-)

Reply:
> NoCompilerInlining is probably least wierd.
> 
> This is presumably expected to operate alongside the MethodImplAttribute.Inline field.  The developer who applies it will be expected to understand the subtlety of Jit inlining and fsc code inlining.
> 
> For sure, it's gonna be wierd to express inlining and no inlining at the same time :-)

Yup, that's my reasoning behind the naming.